From cd83f4095b234652ae65321f35700f7e3441f0f9 Mon Sep 17 00:00:00 2001
From: Ismo Puustinen <ismo.puustinen@intel.com>
Date: Fri, 19 Feb 2016 07:18:28 -0800
Subject: [PATCH 4/6] spi-pxa2xx: fixed ACPI-based enumeration of SPI devices.

Slave devices were not enumerated by ACPI data because the ACPI handle
for the spi-pxa2xx controller was NULL if it was itself enumerated by
PCI.

Original patch by Mika Westerberg <mika.westerberg@intel.com>.

Upstream-status: Inappropriate, will be fixed with a bigger overhaul of
                 SPI/ACPI interaction.

Signed-off-by: Ismo Puustinen <ismo.puustinen@intel.com>
Signed-off-by: Saul Wold <sgw@linux.intel.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 drivers/spi/spi-pxa2xx-pci.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/spi/spi-pxa2xx-pci.c b/drivers/spi/spi-pxa2xx-pci.c
index d19d7f28aecb..562ffc1282bc 100644
--- a/drivers/spi/spi-pxa2xx-pci.c
+++ b/drivers/spi/spi-pxa2xx-pci.c
@@ -169,6 +169,7 @@ static int pxa2xx_spi_pci_probe(struct pci_dev *dev,
 	memset(&pi, 0, sizeof(pi));
 	pi.parent = &dev->dev;
 	pi.name = "pxa2xx-spi";
+	pi.fwnode = dev->dev.fwnode;
 	pi.id = ssp->port_id;
 	pi.data = &spi_pdata;
 	pi.size_data = sizeof(spi_pdata);
-- 
2.5.0

